docs/source/deployment_client.rst

```diff
@@ -0,0 +1,119 @@
+
+.. contents:: Table of Contents
+    :local:
+    :depth: 1
+
+Overview
+--------
+
+The MLflow Deployment Client provides unified CRUD and query APIs to interact with models that are served across various platforms. MLflow provides several built-in deployment clients that provide a consistent and platform-agnostic interface to interact with models served on different platforms, as well as a interface that can be used to implement custom deployment clients via plugins.
+
+
+Builtin Deployment Clients
+--------------------------
+
+The MLflow library provides several built-in deployment clients that can be used to interact with models served on different platforms. The following are the built-in deployment clients provided by MLflow:
+
+* :py:class:`mlflow.deployments.MlflowDeploymentClient`: A deployment client for served models in the MLflow Deployments Server
+* :py:class:`mlflow.deployments.DatabricksDeploymentClient`: A deployment client for served models on Databricks.
+* :py:class:`mlflow.deployments.OpenAIDeploymentClient`: A deployment client for served models on OpenAI or Azure OpenAI platforms.
+
+Deployment Client Factory functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+These clients can be accessed using the following factory methods:
+
+* :py:func:`mlflow.deployments.get_deploy_client`: Returns an instance of the appropriate deployment client based on the specified deployment target.
+* :py:func:`mlflow.deployments.get_deployments_target`: Returns the deployment target associated with the specified deployment client.
+* :py:func:`mlflow.deployments.set_deployments_target`: Sets the deployment target for the specified deployment client. The deployment target can also be set using the ``MLFLOW_DEPLOYMENT_TARGET`` environment variable. If running in Databricks, the deployments target will be set to ``databricks`` by default.
+
+The following is an example of how to create an instance of a Databricks deployment client:
+
+.. code-block:: python
+
+    from mlflow.deployments import get_deploy_client
+
+    client = get_deploy_client("databricks")
+
+Custom Deployment Clients
+-------------------------
+
+MLflow also provides a :py:class:`mlflow.deployments.BaseDeploymentClient` interface that can be used to implement custom deployment clients. These custom deployment clients can be installed as plugins. For more information, see `MLflow Plugins <https://mlflow.org/docs/latest/plugins.html>`_.
+
+Deployment Client Query APIs
+----------------------------
+
+MLflow deployment clients provide a unified query API for deployments or model endpoints. This is the :py:func:`mlflow.deployments.BaseDeploymentClient.predict` API.
+
+For example, to query ``gpt-4`` model endpoint on OpenAI, you can use the following code:
+
+.. code-block:: python
+
+    import os
+    from mlflow.deployments import get_deploy_client
+
+    os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
```

Comment:
nit about defining API KEYS in code - in some other tutorials, we explicitly warn users from doing this. 

For reference, here is how I did it for OpenAI: https://output.circle-artifacts.com/output/job/dc2b2586-38d1-4c99-a00f-4993f6c30ed4/artifacts/0/docs/build/html/llms/openai/notebooks/openai-quickstart.html#API-Key-Security-Overview